### DISCUSSION TITLE
Investigation of need and feasibility of installation directions and bring up guide for affordable 64 Bit Single Board Computers (SBCs)

### DISCUSSION THREAD

**Discussion by hamzah-mujawar** [TheOdinProject/curriculum#29203] [Ideas]
Hi,

I'd like to discuss whether the curriculum could support instructions for bringing up and providing installation directions for a select few 64-bit Single Board Computers (SBCs).

By limiting scope to a few selected and tested boards the bring up and installation directions can be more accurate and will be much less hassle to maintain. 

This list can also be later extended based on user demand and feedback.

### Focus on 64 bit SBCs
I think the best option is to limit the scope to 64-bit boards, this is to ensure compatibility with the software we use in the curriculum, this puts it on par to the standard desktop experience.

### But why?
1. A lot of end users might be on aging hardware which could mean that they might have issues updating to windows 10, using a virtual machine due to limitations in available computational resources, or they might have issues dual booting Ubuntu due to a variety of factors (i.e. Beeping issues due to an adapter warning in one case).
2. Most users have no issue dual booting, using a VM or using WSL. And most users have sufficient available space to be able to do so. If however, they do not space on their drives, and are not able to make space, or not able to install an internal/external drive.
3. Some users prefer to not touch their laptop's BIOS configuration such as disabling safeboot. Or they may have access to only a work laptop that they prefer not to touch. 
4. Some users, might have access to a popular 64-bit SBC.

### Proposal
1. A study into the affordability of 64-bit SBC, this does depend region to region. So users can report the prices of popular 64-bit SBCs which will help narrow the list down. 
2. Affordable 64-bit SBCs such as [Libre Computer Renegade](https://libre.computer/products/roc-rk3328-cc/), [Radxa X4](https://www.aliexpress.com/item/1005007516871199.html). Support Ubuntu 22.04 LTS and provide preinstalled images or documentation: [link to libre computer ubuntu page](https://hub.libre.computer/t/ubuntu-22-04-jammy-lts-for-libre-computer-boards/20), [Raxda X4 docs](https://docs.radxa.com/en/x/x4/getting-started/install-os).
3. I haven't provided a comprehensive list of the best possible SBC for the price as this depends on the region and the availability. But the few that I have mentioned are cheaper than Raspberry pi (again this depends on region).

### What I can do.
1. I can document steps for the finalised list of SBCs. 
2. I can help research and compile a list of SBC's *per region* and select the overall most affordable and convenient one for end-users.

### What do you think?
I personally am very excited to help research and extend support to affordable 64 bit SBCs. The aim is to ensure that it remains fully compatible with the curriculum, and with few moving parts as possible. The main chunk of time should be spent on researching the most cost effective and available SBC per region, as pricing differs by regions, so does availability. Once finalised, the guides can help end-users set up these devices with minimal hassle.

Looking forward to everyone's feedback!

Kind Regards,
Hamzah Mujawar
discord: Madger Badger @ madgerbadger

P.S. Kindly ignore any grammatical mistakes that might have crept it. If I'm not clear, I am happy to re-explain.

**Comment by xandora** [comment]
I have a couple initial thoughts about adding support for SBC devices.

* Who maintains the instructions? While we are an open-source project, the bulk of the work is provided and supported by the core and maintainer teams. Not being able to reliably test the curriculum on these devices is a bit of a concern.
* SBC devices are primarily enthusiast products. The overlap of people using them and *not* being savvy enough to just follow the Linux instructions is going to be pretty miniscule. While we do try to cater the curriculum to as many people as possible, this feels like quite a niche corner of the userbase.

**Reply by hamzah-mujawar** [reply]
Hi,

Firstly, thank you for your input on this matter.

1. Regarding your first point, I agree testing compatibility will be challenging. However, by limiting the scope to just a few (one or two) SBCs which can be thoroughly tested with the curriculum would be maintainable. I am willing to provide and collaborate with the core and maintainer teams to provide relevant details such as testing results, documentation and whatever else is required.
2. I agree with the sentiment that SBC devices are enthusiast products. However, they can also be a cheap way for users to embark on the odin's project curriculum, if they cannot afford name-brand computer. Regarding the point for the instructions, SBC boards can be picked with good documentation (that should take some weight off the maintainers and the core team), many SBC manufacturers provide preinstalled images or have similar installation instructions to your standard laptop.

I don't think the end users or the maintainers should have to worry about providing their own images as SBCs can be picked with long term support from the manufacturers.
